src/NUnitFramework/framework/Internal/Commands/OneTimeSetUpCommand.cs

```diff
@@ -75,15 +77,18 @@ public override TestResult Execute(TestExecutionContext context)
                     Test.Fixture = _suite.Fixture;
                 }
 
-                for (int i = _setUpTearDown.Count; i > 0; )
-                    _setUpTearDown[--i].RunSetUp(context);
+                if (!(suiteResult.Test is TestFixture && !suiteResult.Test.HasChildren))
```

Comment:
Actually `if (suiteResult.Test.HasChildren)` passes all tests too. The rule (that user setup and teardown are not run for 0 child tests) only makes sense in the context of fixtures, right?